examples/29_rgb_depth_aligned.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+import cv2
+import numpy as np
+import depthai as dai
+
+# Optional. If set (True), the ColorCamera is downscaled from 1080p to 720p.
+# Otherwise (False), the aligned depth is automatically upscaled to 1080p
+downscaleColor = True
+
+pipeline = dai.Pipeline()
+
+cam = pipeline.createColorCamera()
+cam.setBoardSocket(dai.CameraBoardSocket.RGB)
+cam.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+if downscaleColor: cam.setIspScale(2, 3)
+# For now, RGB needs fixed focus to properly align with depth.
+# This value was used during calibration
+cam.initialControl.setManualFocus(130)
+
+rgbOut = pipeline.createXLinkOut()
+rgbOut.setStreamName("rgb")
+cam.isp.link(rgbOut.input)
+
+left = pipeline.createMonoCamera()
+left.setResolution(dai.MonoCameraProperties.SensorResolution.THE_720_P)
+left.setBoardSocket(dai.CameraBoardSocket.LEFT)
+
+right = pipeline.createMonoCamera()
+right.setResolution(dai.MonoCameraProperties.SensorResolution.THE_720_P)
+right.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+
+stereo = pipeline.createStereoDepth()
+stereo.setConfidenceThreshold(200)
+# LR-check is required for depth alignment
+stereo.setLeftRightCheck(True)
+stereo.setDepthAlign(dai.CameraBoardSocket.RGB)
+left.out.link(stereo.left)
+right.out.link(stereo.right)
+
+depthOut = pipeline.createXLinkOut()
+depthOut.setStreamName("depth")
+# Currently we use the 'disparity' output. TODO 'depth'
+stereo.disparity.link(depthOut.input)
+
+# Pipeline defined, now the device is connected to
+with dai.Device(pipeline) as device:
+    # Start pipeline
+    device.startPipeline()
+
+    qColor = device.getOutputQueue(name="rgb",   maxSize=4, blocking=False)
+    qDepth = device.getOutputQueue(name="depth", maxSize=4, blocking=False)
+
+    frameRgb = None
+    frameDepth = None
+
+    while True:
+        queueName = device.getQueueEvent(("rgb", "depth"))
```

Comment:
Do the following instead.
If host is too slow, events and messages can queue up.

```
queueEvents = device.getQueueEvents(("rgb", "depth"))
for queueName in queueEvents:
    packets = device.getOutputQueue(queueName).getAll()
    for packet in packets: 
         ...
```

Reply:
Had to change `.getAll()` to `.tryGetAll`, as due to repeated queue names in the returned event list, `getAll()` was blocking, with frames from other queues being received in the meantime, resulting in huge delays between iterations over time:
`logs: time-delta [s] between main loop iterations, followed by queueEvents`
```
tdelta: 0.0 ['rgb']
tdelta: 0.412 ['rgb', 'depth', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'depth']
tdelta: 0.36 ['depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'rgb']
tdelta: 0.498 ['depth', 'depth', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb']
tdelta: 0.394 ['depth', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb']
tdelta: 0.537 ['depth', 'rgb', 'rgb', 'rgb', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'depth', 'rgb', 'rgb']
tdelta: 0.813 ['depth', 'depth', 'rgb', 'rgb', 'depth', 'rgb', 'depth', 'depth', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'rgb']
tdelta: 1.295 ['depth', 'depth', 'depth', 'depth', 'rgb', 'rgb', 'rgb', 'depth', 'depth', 'depth', 'rgb', 'rgb', 'rgb', 'rgb', 'depth', 'rgb', 'rgb', 'depth', 'rgb', 'depth', 'depth', 'depth', 'rgb', 'rgb']
tdelta: 1.342 ['depth', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'rgb', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'depth', 'depth', 'rgb', 'rgb', 'rgb']
...
```

Reply:
@alex-luxonis good catch, `getAll` with `getQueueEvents` wouldn't play well